src/Scope.h

```diff
@@ -20,25 +21,37 @@ class Scope : public BroObj {
 	explicit Scope(ID* id, attr_list* al);
 	~Scope() override;
 
-	ID* Lookup(const char* name) const	{ return local->Lookup(name); }
-	void Insert(const char* name, ID* id)	{ local->Insert(name, id); }
-	ID* Remove(const char* name)
+	ID* Lookup(const std::string& name) const
 		{
-		HashKey key(name);
-		return (ID*) local->Remove(&key);
+		const auto& entry = local.find(name);
+		if ( entry != local.end() )
+			return entry->second;
+
+		return nullptr;
+		}
+	void Insert(const std::string& name, ID* id)	{ local[name] = id; }
```

Comment:
The existing Dict code doesn't warn at all if you try to insert something with the same `HashKey`

Reply:
I guess I'm just thinking that it might be an error to re-assign a name's ID in a scope. I could be wrong, but I'm having trouble thinking of a situation where you'd want to do that. Something like:
```C++
auto where = local.find(name);
if (where != local.end())
    //error / warning
where->second = id;
```
Shouldn't be too much different performance wise and might help catch issues later down the line.